src/Employer/Employer.Web/ViewModels/Validations/ValidGuidAttribute.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Esfa.Recruit.Employer.Web.ViewModels.Validations
+{
+    public class ValidGuidAttribute : ValidationAttribute
+    {
+        public override bool IsValid(object value)
+        {
+            if (value == null)
+            {
+                return true;
+            }
+
+            //Ensuring guid is not 00000000-0000-0000-0000-000000000000

```

Comment:
Good point, but this should resolve to 404 I think. Don;t think it si something that should be validated the way it is as the user can't see or edit the request body.

Reply:
I put this in to make more sense of the validation unit tests.  You could create an edit model in a Test without specifying a VacancyId and it would pass validation which didn't seem right.  I haven't looked at this in the context of the user journey however.